api/desecapi/models.py

```diff
@@ -418,6 +444,83 @@ def save(self, *args, **kwargs):
         self.full_clean(validate_unique=False)
         super().save(*args, **kwargs)
 
+    def clean_records(self, records_presentation_format):
+        """
+        Validates the records belonging to this set. Validation rules follow the DNS specification; some types may
+        incur additional validation rules.
+
+        Raises ValidationError if violation of DNS specification is found.
+
+        Returns a set of records in canonical presentation format.
+
+        :param records_presentation_format: iterable of records in presentation format
+        """
+        rdtype = rdatatype.from_text(self.type)  # TODO may raise if type is unknown
+
+        def _error(record, msg):
+            raise ValidationError(f'Record content of {self.name}/{self.type} invalid: `{record}`: {msg}')
```

Comment:
replaced by `'`

Reply:
I would do: `f'Record content of {self.name}/{self.type} invalid: {msg}. Record: {record}'`

This way, you get rid of any surrounding delimiter, which especially in case of TXT records may be confusing to the user ("why does it show `'`? I thought I passed `"`!" etc.) However, I don't feel strongly.

Reply:
Leading and trailing whitespace may matter in the validation, which is why I included the delimiters. The TXT record would show `'"my record"'`, so that should be comprehensible.